web/index.js

```diff
@@ -12,12 +11,38 @@ var {Main} = require('./app/books/main')
 var {Gallery} = require('./app/books/gallery')
 var {Book} = require('./app/books/book')
 var {About} = require('./app/pages/about')
+var {Login} = require('./app/pages/login')
+var {Signup} = require('./app/pages/signup')
 
 import {assign} from 'lodash'
+import {UserModel} from './app/model'
+import {updateLocalStorage} from './app/helpers'
 
 class App extends React.Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {currentUser: this.props.currentUser}
+  }
+
+  componentDidMount() {
+    UserModel.checkAuth()
+    .then((user) => this.setState({currentUser: user}))
+  }
+
+  setCurrentUser(user) {
+    updateLocalStorage('userToken', user.token)
```

Comment:
Throw this into a client-side model? I'd recommend making a new model file. The current one is getting big.